test/testdata/resolver/optional_nil.rb

```diff
@@ -5,11 +5,13 @@ class Test
   sig {params(x: String).returns(String)}
   def foo(x = nil) # error: Argument does not have asserted type `String`
     x
+  # ^ error: Returning value that does not conform to method result type
```

Comment:
wow why was this error hidden before?

Reply:
The parameter was only initialized with `LoadArg(x)` before, which has the expected type for the parameter. Since the CFG now sees the default and knows that `x` has the value of the argument or the default, we get the error on the return as well 👍